src/Antiforgery/src/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -81,39 +101,80 @@ public AntiforgeryToken Deserialize(string serializedToken)
      *   |    `- Username: UTF-8 string with 7-bit integer length prefix
      *   `- AdditionalData: UTF-8 string with 7-bit integer length prefix
      */
-    private static AntiforgeryToken? Deserialize(BinaryReader reader)
+    private static AntiforgeryToken? Deserialize(ReadOnlySpan<byte> tokenBytes)
     {
+        var offset = 0;
```

Comment:
I don't think there is any reuse of previous sections of `tokenBytes`, so can we get rid of `offset` and use `tokenBytes = tokenBytes.Slice(...)` instead? Offset is just a recipe for mistakes (as we've seen with `ArraySegment`).